### PR TITLE
terminal: Set `TERM_PROGRAM` and `TERM_PROGRAM_VERSION` environment variables in integrated terminal

### LINKED ISSUE #4571: set TERM_PROGRAM environment variable on integrated shell
### Check for existing issues

- [X] Completed

### Describe the feature

Currently the builtin terminal has `ZED_TERM=true` env set, but it is not as universal as `TERM_PROGRAM` and `TERM_PROGRAM_VERSION`. These env vars can be used in shell initialization script for customization. 

`TERM_PROGRAM` and `TERM_PROGRAM_VERSION` are used by vscode, Lens, Apple Terminal, iterm and other terminal programs, some reference links:

1. https://github.com/microsoft/vscode/issues/29426
2. https://code.visualstudio.com/docs/terminal/shell-integration
3. https://apple.stackexchange.com/questions/42714/what-is-the-term-program-and-term-program-version-environment-variables-used-fo
4. https://gitlab.gnome.org/GNOME/console/-/issues/248
5. https://github.com/mintty/mintty/issues/776

Workaround configuration:

```json
{
  "terminal": {
    "env": {
      "TERM_PROGRAM": "zed"
    }
  }
}
```

### If applicable, add mockups / screenshots to help present your vision of the feature

Add the following default env vars to integrated shell:

1. TERM_PROGRAM=zed (or TERM_PROGRAM=Zed if you like)
2. TERM_PROGRAM_VERSION=0.96.4

### PR DESCRIPTION
**Pull request by ldm0**
Release Notes:

- Set `TERM_PROGRAM` and `TERM_PROGRAM_VERSION` environment variables in the integrated terminal ([#4571](https://github.com/zed-industries/zed/issues/4571)).

----
After this PR:

image

These two environment variables are essential when people need to detect terminal type and do something. Many popular terminals set them.

fixes https://github.com/zed-industries/zed/issues/4571

### CODE REVIEW

**Comment by ldm0** [comment]
@cla-bot check

File: crates/terminal/build.rs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+fn main() {
```
Comment (SomeoneToIgnore) [inline crates/terminal/build.rs:1]: We have a https://github.com/zed-industries/zed/blob/d32e9f759c289a111f4a104b40f258c00fd4727b/crates/lsp/src/lsp.rs#L667 with the app version.

does this way not work for the case?

**Review by SomeoneToIgnore — COMMENTED** [review]
Looks good but I think we can avoid dragging in more dependencies and build scripts already.

File: crates/terminal/build.rs (hunk 1 above)
Comment (ldm0) [reply crates/terminal/build.rs]: Thanks for pointing this out, I hadn't noticed this before. Fixed.

**Review by SomeoneToIgnore — APPROVED** [review]
One last bit: `cx` is a last parameter in an overwhelming majority of the codebase, so let's use that.

And also remove the comment about copying out, it's not really needed.

**Comment by ldm0** [comment]
Fixed. Making `cx` the last parameter simplifies the code, which is great. :-)

**Review by SomeoneToIgnore — APPROVED** [review]
Thanks!

**Pull request merged**